pkg/clusteragent/autoscaling/workload/pod_patcher.go

```diff
@@ -100,31 +100,27 @@ func (pa podPatcher) ApplyRecommendations(pod *corev1.Pod) (bool, error) {
 	}
 
 	// Even if annotation matches, we still verify the resources are correct, in case the POD was modified.
+	// K8s guarantees that the name for an init container or normal container are unique among all containers.
+	// It means that dispatching recommendations just by container names is sufficient
 	for _, reco := range autoscaler.ScalingValues().Vertical.ContainerResources {
 		for i := range pod.Spec.Containers {
 			cont := &pod.Spec.Containers[i]
-			if cont.Name != reco.Name {
-				continue
+			if cont.Name == reco.Name {
+				patched = patchContainerResources(reco, cont) || patched
+				break
 			}
-			if cont.Resources.Limits == nil {
-				cont.Resources.Limits = corev1.ResourceList{}
-			}
-			if cont.Resources.Requests == nil {
-				cont.Resources.Requests = corev1.ResourceList{}
-			}
-			for resource, limit := range reco.Limits {
-				if limit != cont.Resources.Limits[resource] {
-					cont.Resources.Limits[resource] = limit
-					patched = true
-				}
-			}
-			for resource, request := range reco.Requests {
-				if request != cont.Resources.Requests[resource] {
-					cont.Resources.Requests[resource] = request
-					patched = true
-				}
+		}
```

Comment:
The function is getting long, what do you think of extracting this as a helper?

Reply:
good idea, done